ui/src/app/shared/test/testsystems/testsystems.spec.ts

```diff
@@ -0,0 +1,518 @@
+
+  private readonly adminAndInstallerGridModal: FormlyFieldConfig[] = [{ "key": "ems30048", "type": "input", "templateOptions": { "attributes": { "title": "Netz" }, "required": true, "options": [{ "lines": [{ "type": "line", "name": "Keine Netzverbindung!", "channel": "_sum/GridMode", "channelCondition": 2 }, { "type": "line", "name": "Bezug", "channel": "_sum/GridActivePower" }, { "type": "line", "name": "Einspeisung", "channel": "_sum/GridActivePower" }, false, { "type": "line", "name": "Phase L1 ", "indentation": "5%", "channel": "meter0/ActivePowerL1", "children": [{ "type": "line-item", "channel": "meter0/VoltageL1", "indentation": "5%" }, { "type": "line-item", "channel": "meter0/CurrentL1", "indentation": "5%" }, { "type": "line-item", "channel": "meter0/ActivePowerL1", "indentation": "5%" }] }, { "type": "line", "name": "Phase L2 ", "indentation": "5%", "channel": "meter0/ActivePowerL2", "children": [{ "type": "line-item", "channel": "meter0/VoltageL2", "indentation": "5%" }, { "type": "line-item", "channel": "meter0/CurrentL2", "indentation": "5%" }, { "type": "line-item", "channel": "meter0/ActivePowerL2", "indentation": "5%" }] }, { "type": "line", "name": "Phase L3 ", "indentation": "5%", "channel": "meter0/ActivePowerL3", "children": [{ "type": "line-item", "channel": "meter0/VoltageL3", "indentation": "5%" }, { "type": "line-item", "channel": "meter0/CurrentL3", "indentation": "5%" }, { "type": "line-item", "channel": "meter0/ActivePowerL3", "indentation": "5%" }] }, { "type": "line-info", "name": "Die Summe der einzelnen Phasen kann aus technischen Gründen geringfügig von der Gesamtsumme abweichen." }] }] }, "wrappers": ["formly-field-modal"] }]
+
+  public widgets: Widget[] = [
+    new Grid(
+      new Modal("GridModal", new Map()
+        .set(Role.ADMIN, this.adminAndInstallerGridModal)
+        .set(Role.INSTALLER, this.adminAndInstallerGridModal)
+        .set(Role.OWNER, this.ownerAndGuestGridModal)
+        .set(Role.GUEST, this.ownerAndGuestGridModal)
+      ))];
+
+  constructor() {
+    super();
+    super.setEdge(new Edge(this.key, "", "", "2023.3.5", Role.ADMIN, true, new Date()));
+    super.setConfig(this.components, this.factories);
+  }
+}
+
+describe('ExampleSystemsTest', () => {
+
+  let translate: TranslateService;
+  beforeEach((() => {
+    TestBed.configureTestingModule({
+      imports: [
+        TranslateModule.forRoot({ loader: { provide: TranslateLoader, useClass: MyTranslateLoader, }, defaultLanguage: 'de' }),
```

Comment:
you can use the constant `Language.DEFAULT.key`

Reply:
sure